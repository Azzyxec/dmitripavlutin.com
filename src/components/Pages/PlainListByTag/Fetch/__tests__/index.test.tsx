import * as React from 'react';
import { shallow } from 'enzyme';

import PlainListByTagFetch from '../index';

const props = {
  pageContext: {
    tag: 'javascript',
  },
  data: {
    allMarkdownRemark: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Useful techniques to facilitate React server-side rendering',
              description: 'Useful techniques to facilitate React server-side rendering.',
              published: '2018-03-17',
              slug: 'useful-techniques-react-server-side-rendering',
              tags: ['react', 'server side rendering'],
            },
          },
        },
      ],
    },
  },
};

describe('<PlainListByTagFetch />', function() {
  it('should render plain list by tag template', function() {
    const wrapper = shallow(<PlainListByTagFetch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
