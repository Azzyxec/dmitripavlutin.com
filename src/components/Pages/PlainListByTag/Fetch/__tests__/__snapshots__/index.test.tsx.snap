// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainListByTagFetch /> should render plain list by tag template 1`] = `
<PlainListByTagTemplate
  posts={
    Array [
      Object {
        "description": "Useful techniques to facilitate React server-side rendering.",
        "published": "2018-03-17",
        "slug": "useful-techniques-react-server-side-rendering",
        "tags": Array [
          "react",
          "server side rendering",
        ],
        "title": "Useful techniques to facilitate React server-side rendering",
      },
    ]
  }
  tag="javascript"
/>
`;
